### ISSUE TITLE
Duplicate ID in inheritance hierarchy. (0x800288C6 (TYPE_E_DUPLICATEID))

### ISSUE THREAD

**Issue by mrinaldhawan** [dspace-group/dscom#317]
I am currently working on migration of the several projects from framework 4.8 to .Net 8.0. I have a couple of projects where I have to add multiple COM References.  When I use the dscom32 utility to generate tlb files, I get the following error.

Duplicate ID in inheritance hierarchy. (0x800288C6 (TYPE_E_DUPLICATEID))

I have created a sample code for you to see the issue, which replicates same behavior, and the attached the code as a zip. 
[SampleLibrary.zip](https://github.com/user-attachments/files/18138814/SampleLibrary.zip)

dscom32 tlbexport D:\SampleLibrary\SampleLibrary\bin\x86\Debug\net8.0\SampleLibrary.dll
Failed to export type library. Failed to set GUID for _Chart. Duplicate ID in inheritance hierarchy. (0x800288C6 (TYPE_E_DUPLICATEID))

Tried using the option  '--createmissingdependenttlbs false'. I still get the same error.

Keeping "EmbedInteropTypes" false, the regular build generates Interop.Microsoft.Office.Interop.Excel.  
After some tweaking, I realised, if "EmbedInteropTypes" is marked false, dscom32 utility generates the tlb file. 

My main concern is, I will have to package the dll and ensure that it is present on the End User's machine.  So I actually want to generate tlb file with "EmbedInteropTypes" true. How can I do this? Am I missing something here or this requires a addition to the dscom utility code.

**Comment by SOsterbrink** [comment]
Just to be sure: You are switching all related libraries at once from 4.8 to 8.0 or are you doing it one by one? 
The error message reminds me of #296 

It's unlikely I will find time to look into it this year because I'll be on x-mas vacation

**Comment by matthiasnissen** [comment]
@mrinaldhawan I had a closer look at your uploaded project. By embedding the interop types your assembly contains the namespace `Microsoft.Office.Interop.Excel` with the CoClass `Chart` and the interface `_Chart`. Strangely both types are attributed with the same GUID attribute. The GUID of both types is the GUID of  `_Chart` interface from the Excel TLB. 

Your code `var chart = new Chart();` is translated to 
`Chart chart = (Chart)Activator.CreateInstance(Marshal.GetTypeFromCLSID(new Guid("00020821-0000-0000-C000-000000000046")));` which is the correct GUID of the CoClass from the Excel TLB.

As I understand it, the embedding ensured that a PIA is no longer needed. The fact that dscom also packs the embedded types into the TLB to be generated is therefore unnecessary, as the Excel TLB already exists on the system. dscom should ignore the imported types on TLB creation.

**Comment by marklechtermann** [comment]
Fixed with Release: [https://github.com/dspace-group/dscom/releases/tag/v1.13.0](https://github.com/dspace-group/dscom/releases/tag/v1.13.0)

**Comment by mrinaldhawan** [comment]
First of all, sorry for a delayed response. I was involved in some other work, and couldn't participate actively.

@matthiasnissen The fix works like a charm. Thank you so much for looking into this. 
I tested it on both the sample app with Excel dll, and another Interop dll (Properietory), and it works like a charm. 

I think I found an official link by Microsoft which details this behavior. 
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.coclassattribute?view=net-9.0#remarks

@SOsterbrink To answer your question, I had actually migrated all the projects to .Net 8.0. Dscom32 utility worked perfectly in projects where COMReferences were not added. But then later, I realized about this issue. After fiddling with some project attributes and reading about them, specifically EmbedInteropTypes, I figured this needs to be an added functionality in this utility, and approached you guys.

Thanks again team for actively looking into these issues and making our life easier! Cheers...
